### ISSUE TITLE
Trusted peer mode

### ISSUE THREAD

**Issue by cpacia** [gcash/neutrino#10]
This feature isn't that import now while the connections are not encrypted or authenticated, but there's some talk about enabling QUIC as an experiment in BCHD which would give us encrypted and authenticated connections. 

Once that is in place it would be nice to have a "trusted peer" mode where we can connect the neutrino client to a node of our choice and give it an authentication key to authenticate against. 

Once this is done it would likely make sense to pack the bloom filter we are using for the mempool with our actual watch addresses rather than using a match all bloom filter like we currently do. This would prevent us from using unnecessary bandwidth as we don't care about wire privacy when connected to our own node.
